spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -78,13 +73,10 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		ConsumerProperties consumerProperties =
 				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName);
 		if (binder instanceof ExtendedPropertiesBinder) {
-			ExtendedPropertiesBinder extendedPropertiesBinder = (ExtendedPropertiesBinder) binder;
-			Object extension = extendedPropertiesBinder.getExtendedConsumerProperties(inputChannelName);
-			ExtendedConsumerProperties extendedConsumerProperties = new ExtendedConsumerProperties(extension);
-			BeanUtils.copyProperties(consumerProperties, extendedConsumerProperties);
-			consumerProperties = extendedConsumerProperties;
+			consumerProperties = ProducerConsumerPropertiesUtil.getExtendedConsumerProperties((ExtendedPropertiesBinder) binder,
```

Comment:
Instead of making this a static method on an external class, can we just encapsulate this as part of `ChannelBindingService` ?